examples/point_api_last_full_issuance.py

```diff
@@ -0,0 +1,110 @@
+"""
+Example of how to use the Spire Weather Point API to download forecast data for a particular product
+bundle and time bundle. This code is set up to download the latest full forecast issuance,
+thereby ignoring the most recent forecast if it is incomplete and still populating the API.
+"""
+import argparse
+import os
+from urllib.parse import urljoin
+from datetime import timedelta
+import dateutil.parser
+import requests
+
+HOST = 'https://api.wx.spire.com'
+API_KEY = os.getenv('spire-api-key')
+
+
+def make_request(api_key, lat, lon, time_bundle, issuance_time):
+    # Build the URL, add the headers and query parameters.
+    url = urljoin(HOST, '/forecast/point')
+    params = {
+        'bundles': 'basic',
+        'time_bundle': time_bundle,
+        'lat': lat,
+        'lon': lon
+    }
+    # Add the `issuance_time` parameter if it is specified
+    if issuance_time != None:
+        params['issuance_time'] = issuance_time
+    # Set the auth header
+    headers = {'spire-api-key': api_key}
+    # Make the API request
+    response = requests.get(url, headers=headers, params=params)
+    # Return the API response
+    return response
+
+
+def get_response_data(response):
+    # Convert the raw response into JSON format
+    json_response = response.json()
+    # Ensure the 'data' field is present
+    if 'data' not in json_response:
+        raise Exception('Response did not contain a data element', json_response)
+    # Return the response data
+    return json_response['data']
+
+
+def get_expected_forecast_size(time_bundle):
```

Comment:
If I was an end user, I might be more interested in specific times, rather than the count of lead times.  Though, I don't know if that would fit into what you are thinking here.

My thought would be to change the idea "Do I have 25 lead times" to "Do I have 2020-05-22T00:00:00 and , 2020-05-23T06:00:00" possibly including all lead times from an issuance.

This could also be brittle with the assumption of lead times per bundle.  If we ever change the bundle definitions (which might be rare but possible) then this code would be broken.  I don't have a suggestion for that at the moment other than we build the get_last_complete_issuance logic into the backend where we control the time_bundle definitions.